src/app/trip-page/trip-photo-full-view/trip-photo-full-view.component.ts

```diff
@@ -0,0 +1,61 @@
+import { ChangeDetectorRef, Component, Input, OnChanges, OnInit } from '@angular/core';
+import { BehaviorSubject, Observable, of, Subject, take } from 'rxjs';
+import { TripPhoto } from 'src/app/shared';
+
+@Component({
+  selector: 'app-trip-photo-full-view',
+  templateUrl: './trip-photo-full-view.component.html',
+  styleUrls: ['./trip-photo-full-view.component.css']
+})
+export class TripPhotoFullViewComponent implements OnChanges {
+  initTrip: TripPhoto = {
+    id: '',
+    tripId: 0,
+    fileName: '',
+    fileType: '',
+    path: '',
+    addedByUser: '',
+    alt: '',
+  }
+  @Input() displayFullView$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
```

Comment:
Passing a BehaviorSubject into a component is not a best practice. Depending on what you are doing with this, you should look to inject a service into the constructor that calls a method that would next onto your BehaviorSubject. You always want to protect your BehaviorSubjects. In almost all cases they should be private. It's ok to create small services that allow a few close components to interact with each other.